mobile-app/cypress/e2e/functional/wallet/settings/settings.spec.ts

```diff
@@ -227,3 +229,167 @@ context('Wallet - Settings - Address Book', () => {
+context('Wallet - Settings - Service Provider', () => {
+  before(() => {
+    cy.createEmptyWallet(true)
+    cy.getByTestID('header_settings').click()
+  })
+
+  defichainUrlEnvs.forEach((defichainUrlEnv) => {
+    const url = defichainUrls[defichainUrlEnv]
+    it(`should display default on first app load on ${defichainUrlEnv}`, () => {
+      cy.intercept('**/settings/flags', {
+        body: [
+          {
+            id: 'service_provider',
+            name: 'Service Provider',
+            stage: 'beta',
+            version: '>0.0.0',
+            description: 'Allows the usage of custom server provider url',
+            networks: [EnvironmentNetwork.RemotePlayground, EnvironmentNetwork.LocalPlayground, EnvironmentNetwork.MainNet, EnvironmentNetwork.TestNet],
+            platforms: ['ios', 'android', 'web']
+          }
+        ]
+      })
+      localStorage.setItem('WALLET.ENABLED_FEATURES', '["service_provider"]')
+      cy.getByTestID('button_selected_network').click().wait(50)
+      cy.getByTestID(`button_network_${defichainUrlEnv}`).click()
+      cy.on('window:confirm', () => {
+      })
+      cy.createEmptyWallet(true).wait(2000)
+      cy.getByTestID('header_settings').click().wait(50)
+      cy.getByTestID('setting_navigate_service_provider').contains('Default')
+      cy.url().should('include', 'app/Settings', () => {
+        expect(localStorage.getItem('WALLET.SERVICE_PROVIDER_URL')).to.eq(url.default)
+      })
+    })
+
+    it(`should have default service provider url on ${defichainUrlEnv}`, () => {
+      cy.getByTestID('setting_navigate_service_provider').click()
+      cy.getByTestID('endpoint_url_input').should('have.value', url.default)
+    })
+
+    it(`input should be locked and not editable on ${defichainUrlEnv}`, () => {
+      cy.getByTestID('endpoint_url_input').should('have.attr', 'readonly')
+    })
+
+    it('can unlock to change service provider endpoint', () => {
+      cy.getByTestID('unlock_button').click()
+      cy.getByTestID('unlock_button').should('not.exist')
+      cy.getByTestID('reset_button').should('exist')
+      cy.getByTestID('endpoint_url_input').should('not.have.attr', 'readonly')
+      cy.getByTestID('button_submit').should('have.attr', 'aria-disabled')
+    })
+
+    it('should type invalid custom provider URL', () => {
+      cy.getByTestID('endpoint_url_input').should('have.value', '')
+      cy.getByTestID('endpoint_url_input').type('http://invalidcustomURL.com')
+      cy.getByTestID('endpoint_url_input_error').contains('Invalid URL')
+      cy.getByTestID('button_submit').should('have.attr', 'aria-disabled')
+    })
+
+    it('should submit valid custom service provider', () => {
+      cy.getByTestID('endpoint_url_input').clear().type(url.custom).wait(3000)
+      cy.getByTestID('button_submit').click().wait(1000)
+      cy.getByTestID('pin_authorize').type('000000').wait(5000)
+      cy.go('back')
+    })
+
+    // it('should display Custom on Server', () => {
```

Comment:
should we remove?

Reply:
Updated. Added a comment on why it's commented

Reply:
To follow fix for this issue ^ 
![image](https://user-images.githubusercontent.com/13292662/174717730-9a87fd96-2d6c-4260-94b8-d70c5e913a33.png)

Reply:
Fixed!